aspnetcore/fundamentals/startup.md

```diff
@@ -1,75 +1,106 @@
+
+Each `IStartupFilter` implements one or more middlewares in the request pipeline. The middlewares are invoked in the order that the `IStartupFilter` implementations are added to the service container.
+
+The [sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/startup/sample/) ([how to download](xref:tutorials/index#how-to-download-a-sample)) demonstrates how to register a middleware with `IStartupFilter`. The sample app includes a middleware that sets an options value from a query string parameter:
+
+[!code-csharp[Main](startup/sample/RequestSetOptionsMiddleware.cs?name=snippet1)]
+
+The `RequestSetOptionsMiddleware` is configured in the `RequestSetOptionsStartupFilter` class:
+
+[!code-csharp[Main](startup/sample/RequestSetOptionsStartupFilter.cs?name=snippet1&highlight=7)]
+
+The `IStartupFilter` is registered in the service container in `ConfigureServices`:
+
+[!code-csharp[Main](startup/sample/Startup.cs?name=snippet1&highlight=3)]
+
+When the app is run and a query string parameter for `option` is provided, the middleware processes the value assignment before the MVC middleware renders the response:
+
+![Browser window showing the rendered Index page. The value of Option is rendered as 'From Middleware' based on requesting the page with the query string parameter and value of option set to 'From Middleware'.](startup/_static/index.png)
 
-For more information about how to use `IApplicationBuilder`, see [Middleware](xref:fundamentals/middleware).
+Middleware execution order is set by the order of `IStartupFilter` registrations:
 
-Additional services, like `IHostingEnvironment` and `ILoggerFactory` may also be specified in the method signature, in which case these services will be [injected](dependency-injection.md) if they are available. 
+* Multiple `IStartupFilter` implementations may interact with the same objects. If ordering is important, order their `IStartupFilter` service registrations to match the order that their middlewares should run.
+* Libraries may add middleware with one or more `IStartupFilter` implementations that run before or after other app middleware registered with `IStartupFilter`. To invoke an `IStartupFilter` middleware before a middleware added by a library's `IStartupFilter`, position the service registration before the library is added to the service container. To invoke it afterward, position the service registration after the library is added.
+* Middleware specified with `IStartupFilter` may short-circuit requests before higher priority middleware runs. Register the `IStartupFilter` for the short-circuited middleware earlier in the service registrations to ensure that it runs before short-circuiting can occur.
```

Comment:
This isn't specific to startup filters, this is a generic statement about middleware ordering.

Reply:
The concept I failed to convey is that libraries that add filters may short-circuit. However, I think the 2nd bullet covers that really, so let's kill this one off. :gun: